daemon/api.go

```diff
@@ -2352,6 +2360,80 @@ func postCreateUser(c *Command, r *http.Request, user *auth.UserState) Response
 	}, nil)
 }
 
+type postUserRemoveData struct {
+	Username string `json:"username"`
+}
+
+func postRemoveUser(c *Command, r *http.Request, user *auth.UserState) Response {
+	_, uid, _, err := postRemoveUserUcrednetGet(r.RemoteAddr)
+	if err != nil {
+		return BadRequest("cannot get ucrednet uid: %v", err)
+	}
+	if uid != 0 {
+		return BadRequest("cannot use remove-user as non-root")
+	}
+
+	var removeData postUserRemoveData
+
+	decoder := json.NewDecoder(r.Body)
+	if err := decoder.Decode(&removeData); err != nil {
+		return BadRequest("cannot decode remove-user data from request body: %v", err)
+	}
+
+	// verify request
+	st := c.d.overlord.State()
+	st.Lock()
+	users, err := auth.Users(st)
+	st.Unlock()
+	if err != nil {
+		return InternalError("cannot get user count: %s", err)
+	}
+
+	username := removeData.Username
+	if username == "" {
+		return BadRequest("cannot remove user: 'username' field is empty")
```

Comment:
Another nitpick to make this slightly less technical: `cannot remove user, no user specified`?

Reply:
Agree that it's a better message, but in user creation we have "cannot create user: 'email' field is empty". Again, should I change in both places to keep it nice and consistent?

Reply:
Go for it :)